OMCompiler/Compiler/BackEnd/Matching.mo

```diff
@@ -5320,34 +5320,29 @@ public function PFPlusExternal
   output BackendDAE.EqSystem osyst;
   output BackendDAE.Shared oshared;
   output BackendDAE.StructurallySingularSystemHandlerArg outArg;
+protected
+  Integer nvars,neqns;
 algorithm
+  neqns := BackendDAEUtil.systemSize(isyst);
+  nvars := BackendVariable.daenumVariables(isyst);
   (osyst,oshared,outArg) :=
   matchcontinue (isyst,ishared,clearMatching,inMatchingOptions,sssHandler,inArg)
```

Comment:
Wasn't the point of introducing `guard` to change this to `match`? did it not work?

Reply:
I moved these two lines in front of the match(continue) because these calculations are done twice if the first case fails.
We could try to convert `matchcontinue` to  `match`, but then we would have to remove the error message. I wanted to be somewhat minimal in the change.

Reply:
Yea i see that, also i did not realize that this is the default matching algorithm, so it would be quite the substantial change.